File: output/gpt_4o/avatar/algo_based_translation/python/java/codeforces_474_B.java

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class codeforces_474_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Step 3: Read input from the user (not used in logic)
        scanner.nextLine();

        // Step 3: Read second input from the user, split it by spaces into a list of strings
        String[] inputValues = scanner.nextLine().split(" ");
        
        // Step 4: Logic Flow
        List<Integer> L = new ArrayList<>();
        int r = 1;
        
        for (String k : inputValues) {
            int count = Integer.parseInt(k);
            for (int i = 0; i < count; i++) {
                L.add(r);
            }
            r++;
        }
        
        // Step 4: Read another input from the user (not used in logic)
        scanner.nextLine();

        // Step 4: Read a final input from the user, split it by spaces into a list of strings
        String[] indices = scanner.nextLine().split(" ");
        
        // Step 6: Data Processing Logic and Output
        for (String j : indices) {
            int index = Integer.parseInt(j) - 1;
            System.out.println(L.get(index));
        }
        
        scanner.close();
    }
}

File: output/gpt_4o/avatar/direct_translation/java/python/codeforces_474_B.py

def fun(arr, q, n, m):
    res = 0
    i, j = 0, n
    while i <= j:
        md = i + (j - i) // 2
        if arr[md] == q:
            return md
        elif arr[md] > q:
            res = md
            j = md - 1
        else:
            i = md + 1
    return res

def main():
    import sys
    input = sys.stdin.read
    data = input().split()
    
    index = 0
    n = int(data[index])
    index += 1
    arr = [0] * n
    arr[0] = int(data[index])
    index += 1
    
    for i in range(1, n):
        x = int(data[index])
        index += 1
        arr[i] = x + arr[i - 1]
    
    m = int(data[index])
    index += 1
    q = [0] * m
    for i in range(m):
        q[i] = int(data[index])
        index += 1
    
    for k in range(m):
        print(fun(arr, q[k], n, m) + 1)

if __name__ == "__main__":
    main()

